r2e_test_server/cli.py

```diff
@@ -1,35 +1,164 @@
-import typer
-import rpyc
-import json
-from r2e_test_server.server import R2EService
-from r2e_test_server.server import start_server
+from typing import List, Optional
+import rpyc, fire
+from r2e_test_server.server import R2EService, start_server
 
+def retrieve_conn(host, port):
+    return rpyc.connect(host, port)
 
-app = typer.Typer(name="r2e-test-server")
+class R2EServer:
 
+    def start(self,
+              repo_id: str = 'zeta',
+              port: int = 3006, 
+              repo_dir: str = '/repos',
+              result_dir: str = '/results',
+              verbose: bool = False):
+        print(f"Starting R2E server on port {port}...")
+        start_server(port=port, 
+                     repo_id=repo_id,
+                     repo_dir=repo_dir, 
+                     result_dir=result_dir,
+                     verbose=verbose)
 
-@app.command()
-def start(
-    port: int = typer.Option(3006, help="Port number to start the R2E server on.")
-):
-    """
-    Starts the R2E server on the specified port.
-    """
-    typer.echo(f"Starting R2E server on port {port}...")
-    start_server(port)
+    def register_fut(self, fut_id: str, module_path: str,
+                     host: str = 'localhost',
+                     port: int = 3006):
+        service: R2EService = retrieve_conn(host, port).root
+        print(service.register_fut(fut_id, module_path))
 
+    def tests(self, 
+              host: str = 'localhost',
+              port: int = 3006):
+        service: R2EService = retrieve_conn(host, port).root
+        R2EServer.Helper.print_result("tests", service.get_tests())
 
-@app.command()
-def stop(host: str = typer.Option("localhost"), port: int = typer.Option(3006)):
-    """
-    Stops the R2E server.
-    """
-    typer.echo("Stopping R2E server...")
-    conn = rpyc.connect(host, port)
-    service = conn.root
-    service.stop_server()
-    conn.close()
+    def patchs(self, 
```

Comment:
typo: patches? also should we call this futs since it calls get_futs? (might be missing something)

Reply:
So I decided to separate the two things, FUTs should be the name of the function, and patch should refer to different version of it. `get_futs` is just historical relic from before I made that decision.

Reply:
got it!